api/http/options.go

```diff
@@ -7,16 +7,51 @@ package http
 
 import (
 	"crypto/tls"
+	"net"
+	"net/http"
+	"time"
 )
 
 // Options holds http configuration properties for communicating with InfluxDB server
 type Options struct {
+	// HTTP client. Default is http.DefaultClient.
+	httpClient *http.Client
 	// TLS configuration for secure connection. Default nil
 	tlsConfig *tls.Config
 	// HTTP request timeout in sec. Default 20
 	httpRequestTimeout uint
 }
 
+// HTTPClient returns the http.Client that is configured to be used
+// for HTTP requests. It will return the one that has been set using
+// SetHTTPClient or it will construct a default client using the
+// other configured options.
+func (o *Options) HTTPClient() *http.Client {
+	if o.httpClient != nil {
+		return o.httpClient
+	}
+	return &http.Client{
+		Timeout: time.Second * time.Duration(o.HTTPRequestTimeout()),
+		Transport: &http.Transport{
+			DialContext: (&net.Dialer{
+				Timeout: 5 * time.Second,
+			}).DialContext,
+			TLSHandshakeTimeout: 5 * time.Second,
+			TLSClientConfig:     o.TLSConfig(),
+		},
+	}
```

Comment:
Why not setting newly created HTTP client to `o.httpClient`? It would allow returning the HTTP client used by the library.

Reply:
I can change it to do that.